Week2/homework/maartjes-work.js

```diff
@@ -46,16 +46,16 @@ const maartjesTasks = monday.concat(tuesday);
 const maartjesHourlyRate = 20;
 
 function computeEarnings(tasks, hourlyRate) {
-  // Replace this comment and the next line with your code
-  console.log(tasks, hourlyRate);
+  const durationInHours = tasks.map(minute => minute.duration / 60);
+  const moreThan2Hours = durationInHours.filter(newArray => newArray >= 2);
+  const wageForTotalHours = moreThan2Hours.map(wage => wage * hourlyRate);
+  const totalWage = wageForTotalHours.reduce((total, num) => total + num);
```

Comment:
As you can see it is a hard job trying to invent accurate names for the variables holding the intermediate results. We can escape from this burden by using a method chain:

```js
function computeEarnings(tasks, hourlyRate) {
  return tasks
    .map(task => task.duration / 60)
    .filter(hour => hour >= 2)
    .map(hour => hour * hourlyRate)
    .reduce((total, amount) => total + amount, 0);
}
```

Now we have a concise, readable piece of code, with accurate names.

Note that storing the intermediate results into separate variables is a good approach if you are debugging, because once you have them in a method chain it becomes harder to debug.

Reply:
That's a good idea that we can use all of these functions (map, filter, map, reduce) in a chain without using a variable, and I learned how to use it now. Thank you very much for your useful and instructive comments.